csrc/root_domain_map.cpp

```diff
@@ -182,6 +183,16 @@ std::unordered_map<IterDomain*, IterDomain*> PairwiseRootDomainMap::map(
       continue;
     }
 
+    // Condition 5
+    // At least one ID is symbolic.
+    // Map producer to consumer if and only if their extents are identical
+    if ((producer_id->isSymbolic() || consumer_id->isSymbolic()) &&
+        (!producer_id->extent()->sameAs(consumer_id->extent()))) {
+      itc++;
+      itp++;
+      continue;
+    }
```

Comment:
This replaces earlier iterations on handling exact mapping of Symbolic domains, which I think were more cumbersome. The meaning of a `Symbolic` IterType is that it could potentially be a `Broadcast` domain. This is essentially reflecting that the extent expression hasn't been evaluated yet. If we have identical extents for two aligned `Symbolic` IterDomains, this should mean they will have the same IterType after concretization. In that case, they should be exact mapped. This lets us propagate bound extents in `ExpressionEvaluator` much more easily, while still avoiding exact mapping between transformed symbolic domains.